### DISCUSSION TITLE
Zephyr SDK and Compilation Discussions

### DISCUSSION THREAD

**Discussion by RohitSrinivasRG** [Mindgrove-Technologies/zephyr#6] [Q&A]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by santhoshm1999** [comment]
@M-Kapil-Shyam 
I'm facing the following problem while compiling tests from zephyr using zephyr sdk

<img width="1842" height="509" alt="image" src="https://github.com/user-attachments/assets/6eb667b3-d2c8-4ad9-bb11-debd6eb5404d" />

<img width="1842" height="509" alt="image" src="https://github.com/user-attachments/assets/898ceafa-02ca-4881-8f48-4c14295464b5" />

It is showing __ABI is incompatible with that of the selected emulation__ and __target emulation \`elf32-littleriscv' does not match \`elf64-littleriscv'__

**Reply by RohitSrinivasRG** [reply]
@santhoshm1999 I think you can create a new discussion instead of using this "Welcome" discussion page.

**Reply by santhoshm1999** [reply]
Yeah, Sure

**Comment by M-Kapil-Shyam** [comment]
The Zephyr SDK has the March of rv32imac by default, and the Zephyr specific picolibc also has rv32imac as default. So in the <zephyr-path/to/zsdk>/riscv64-unknown-elf/<.o,.a> must be replaced with the arch specific .o,.a file which will be present in <zephyr-path/to/zsdk>/riscv64-unknown-elf/<march-you-need>/<version>/medany/<.a,.o>. Same to be done for picolibc's libgcc.a etc also.

**Reply by santhoshm1999** [reply]
Thanks

**Reply by Suma2261** [reply]
<img width="512" height="168" alt="image" src="https://github.com/user-attachments/assets/15f2b885-dd95-423b-96c6-13ce23911f7e" />

 out of these medany folder which is the one we need to use

**Reply by Suma2261** [reply]
I copied it from 'rv64imafdc_zicsr_zifencei' folder
```
cp zephyr-sdk-0.17.2/riscv64-zephyr-elf/picolibc/riscv64-zephyr-elf/lib/rv64imafdc_zicsr_zifencei/lp64d/medany/*  zephyr-sdk-0.17.2/riscv64-zephyr-elf/picolibc/riscv64-zephyr-elf/lib/.
cp zephyr-sdk-0.17.2/riscv64-zephyr-elf/lib/gcc/riscv64-zephyr-elf/12.2.0/rv64imafdc_zicsr_zifencei/lp64d/medany/* zephyr-sdk-0.17.2/riscv64-zephyr-elf/lib/gcc/riscv64-zephyr-elf/12.2.0/.
cp zephyr-sdk-0.17.2/riscv64-zephyr-elf/picolibc/lib/gcc/riscv64-zephyr-elf/12.2.0/rv64imafdc_zicsr_zifencei/lp64d/medany/* zephyr-sdk-0.17.2/riscv64-zephyr-elf/picolibc/lib/gcc/riscv64-zephyr-elf/12.2.0/.
```

**Comment by nithish-mindgrove** [comment]
```
WORK_DIR=                     #your work dir path
SDK_ZEPHYR=                #path to install zephyr sdk

#install dependencies
pip install west
pip install pykwalify

#download zephyr-sdk
cd $SDK_ZEPHYR
wget https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v0.17.4/zephyr-sdk-0.17.4_linux-x86_64.tar.xz

tar -xvf zephyr-sdk-0.17.4_linux-x86_64.tar.xz
rm zephyr-sdk-0.17.4_linux-x86_64.tar.xz
cd zephyr-sdk-0.17.4
./setup.sh -h -c
cd ../
cp zephyr-sdk-0.17.4/riscv64-zephyr-elf/picolibc/riscv64-zephyr-elf/lib/rv64imafdc_zicsr_zifencei/lp64d/medany/*  zephyr-sdk-0.17.4/riscv64-zephyr-elf/picolibc/riscv64-zephyr-elf/lib/.
cp zephyr-sdk-0.17.4/riscv64-zephyr-elf/lib/gcc/riscv64-zephyr-elf/12.2.0/rv64imafdc_zicsr_zifencei/lp64d/medany/* zephyr-sdk-0.17.4/riscv64-zephyr-elf/lib/gcc/riscv64-zephyr-elf/12.2.0/.
cp zephyr-sdk-0.17.4/riscv64-zephyr-elf/picolibc/lib/gcc/riscv64-zephyr-elf/12.2.0/rv64imafdc_zicsr_zifencei/lp64d/medany/* zephyr-sdk-0.17.4/riscv64-zephyr-elf/picolibc/lib/gcc/riscv64-zephyr-elf/12.2.0/.

#clone core_verif
cd $WORK_DIR
git clone git@github.com:Mindgrove-Technologies/core_verif.git
cd core_verif/
make zephyr
cd zephyrproject/zephyr/
git checkout origin/multicore-x1-v1.1
cd $WORK_DIR



```

**Comment by ssanthoshpavan** [comment]
## Config options

CONFIG_UART_INTERRUPT_DRIVEN=y
CONFIG_SMP=y
CONFIG_MP_MAX_NUM_CPUS=4
CONFIG_TICKET_SPINLOCKS=y

### If the proj.conf has already has CONFIG_MP_MAX_NUM_CPUS=1
Only the below 2 lines should be added

CONFIG_UART_INTERRUPT_DRIVEN=y
CONFIG_TICKET_SPINLOCKS=y
